lib/ansible/modules/cloud/amazon/ec2_ami.py

```diff
@@ -73,61 +60,67 @@
-    required: false
-    default: null
   no_reboot:
     description:
       - Flag indicating that the bundling process should not attempt to shutdown the instance before bundling. If this flag is True, the
         responsibility of maintaining file system integrity is left to the owner of the instance.
-    required: false
     default: no
     choices: [ "yes", "no" ]
   image_id:
     description:
       - Image ID to be deregistered.
-    required: false
-    default: null
   device_mapping:
     version_added: "2.0"
     description:
-      - List of device hashes/dictionaries with custom configurations (same block-device-mapping parameters)
+      - List of device hashes/dictionaries with custom configurations (same block-device-mapping parameters).
       - >
-        Valid properties include: device_name, volume_type, size (in GB), delete_on_termination (boolean), no_device (boolean),
-        snapshot_id, iops (for io1 volume_type)
-    required: false
-    default: null
+        Valid properties include: device_name, volume_type, size/volume_size (in GB), delete_on_termination (boolean), no_device (boolean),
+        snapshot_id, iops (for io1 volume_type), encrypted
   delete_snapshot:
     description:
       - Delete snapshots when deregistering the AMI.
-    required: false
     default: "no"
     choices: [ "yes", "no" ]
   tags:
     description:
       - A dictionary of tags to add to the new image; '{"key":"value"}' and '{"key":"value","key":"value"}'
-    required: false
-    default: null
     version_added: "2.0"
   launch_permissions:
     description:
-      - Users and groups that should be able to launch the AMI. Expects
-        dictionary with a key of user_ids and/or group_names. user_ids should
-        be a list of account ids. group_name should be a list of groups, "all"
-        is the only acceptable value currently.
-    required: false
-    default: null
+      - Users and groups that should be able to launch the AMI. Expects dictionary with a key of user_ids and/or group_names. user_ids should
+        be a list of account ids. group_name should be a list of groups, "all" is the only acceptable value currently.
     version_added: "2.0"
+  image_location:
+    description:
+      - The s3 location of an image to use for the AMI.
+    version_added: "2.5"
+  ena_support:
+    description:
+      - A boolean representing whether enhanced networking with ENA is enabled or not.
+    version_added: "2.5"
+  billing_products:
+    description:
+      - A list of valid billing codes. To be used with valid accounts by aws marketplace vendors.
+    version_added: "2.5"
+  ramdisk_id:
+    description:
+      - The ID of the RAM disk.
+    version_added: "2.5"
+  sriov_net_support:
+    description:
+      - Set to simple to enable enhanced networking with the Intel 82599 Virtual Function interface for the AMI and any instances that you launch from the AMI.
```

Comment:
Do we need both sriov_net_support and ena_support?

Reply:
Not sure exactly - they're both specified as options in the boto3 docs. 
http://boto3.readthedocs.io/en/latest/reference/services/ec2.html

EnaSupport (boolean) --

Specifies whether enhanced networking with ENA is enabled.

SriovNetSupport (string) --

Specifies whether enhanced networking with the Intel 82599 Virtual Function interface is enabled.

Just guessing but one is probably for paravirtual and the other may be for hardware virtualization.

Reply:
Just looked up SRIOV - Single Root IO Virtualization - used for PCIe - in which case they are both required. SRIOV is probably relevant for GPU-class instances, whereas ena is more relevant to high-throughput xlarge type instances.

Reply:
More info:
http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/enhanced-networking.html

Both for networking but for different instance types.
SRIOV runs at 10 GPS and ENA at 25 GPS.

Could possibly use one parameter for both and detect which is relevant to the AMI being created and send the required option to boto3..

Reply:
an AMI can have both `ena-support` and `sriov-net-support` enabled. If your kernel is configured correctly, depending on what instance you start an instance of the AMI on, one or the other is used.

These are both really important switches for hi-performance AWS use. Looking forward to the merge!